applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_contact_adaptative_remeshing.py

```diff
@@ -0,0 +1,162 @@
+            "absolute_convergence_tolerance"   : 1.0e-9,
+            "relative_convergence_tolerance"   : 1.0e-4,
+            "max_number_iterations"            : 10,
+            "origin_variable"                  : "AUGMENTED_NORMAL_CONTACT_PRESSURE",
+            "destination_variable"             : "AUGMENTED_NORMAL_CONTACT_PRESSURE",
+            "integration_order"                : 2
+        }
+        """)
+
+        interface_model_part = computing_model_part.GetSubModelPart("Contact")
+        main_model_part = computing_model_part.GetRootModelPart()
+        main_model_part.RemoveSubModelPart("ComputingContact")
+
+        if interface_model_part.HasSubModelPart("SlaveSubModelPart"):
+            slave_interface_model_part = interface_model_part.GetSubModelPart("SlaveSubModelPart")
+        else:
+            slave_interface_model_part = interface_model_part.CreateSubModelPart("SlaveSubModelPart")
+            KM.FastTransferBetweenModelPartsProcess(slave_interface_model_part, interface_model_part, KM.FastTransferBetweenModelPartsProcess.EntityTransfered.NODES, KM.SLAVE).Execute()
+            KM.FastTransferBetweenModelPartsProcess(slave_interface_model_part, interface_model_part, KM.FastTransferBetweenModelPartsProcess.EntityTransfered.CONDITIONS, KM.SLAVE).Execute()
+        if interface_model_part.HasSubModelPart("MasterSubModelPart"):
+            master_interface_model_part = interface_model_part.GetSubModelPart("MasterSubModelPart")
+        else:
+            master_interface_model_part = interface_model_part.CreateSubModelPart("MasterSubModelPart")
+            KM.FastTransferBetweenModelPartsProcess(master_interface_model_part, interface_model_part, KM.FastTransferBetweenModelPartsProcess.EntityTransfered.NODES, KM.MASTER).Execute()
+            KM.FastTransferBetweenModelPartsProcess(master_interface_model_part, interface_model_part, KM.FastTransferBetweenModelPartsProcess.EntityTransfered.CONDITIONS, KM.MASTER).Execute()
+
+        mortar_mapping = KM.SimpleMortarMapperProcess(slave_interface_model_part, master_interface_model_part, map_parameters)
+        mortar_mapping.Execute()
+
+    def AdaptativeRemeshingRunSolutionLoop(self):
+        """This function executes the solution loop of the AnalysisStage for cases where remeshing may be considered
+
+            Keyword arguments:
+            self It signifies an instance of a class.
+        """
+
+        # If we remesh using a process
+        computing_model_part = self.analysis._GetSolver().GetComputingModelPart()
+        root_model_part = computing_model_part.GetRootModelPart()
+
+        while self.analysis.KeepAdvancingSolutionLoop():
+            self.analysis.time = self.analysis._GetSolver().AdvanceInTime(self.analysis.time)
+            # We reinitialize if remeshed previously
+            if root_model_part.Is(KM.MODIFIED):
+                self.analysis.ReInitializeSolver()
+            self.ExecuteBeforeFinalizeSolutionStep()
```

Comment:
Should this not be 'ExecuteBefore**Initialize**SolutionStep()'?

Reply:
True; in fact the method is duplicated with the base class, I remove it

Reply:
And I added the missing method